app/controllers/api/projects_controller.rb

```diff
@@ -84,4 +115,17 @@ def find_project_as_json_with_dependencies!(platform, name, version_name, subset
     project_json
   end
 
+  def date_parameter(symbol)
+    s = params.fetch(symbol, nil)
+    if s.nil?
+      nil
+    else
+      begin
+        DateTime.iso8601(s)
+      rescue ArgumentError
+        raise ActionController::BadRequest.new("bad date parameter")
+      end
+    end
+  end
```

Comment:
Do we gain anything by not just letting the parse error bubble up?

```suggestion
  def date_parameter(symbol)
    return if params[symbol].blank?
    Time.parse(params[symbol])
```

Reply:
I was thinking return BadRequest rather than a 500, but maybe rails already handles it?

also using iso8601 rather than heuristic parser...

Reply:
Hm, I'm curious if rails does the right thing here

Reply:
It looks like plain DateTime.iso8601 would throw ArgumentError leading to a 500